### PR TITLE
Switch to redoc with custom theme

### PR DESCRIPTION
**Pull request by dctanner**
# Pull Request

## Description

We'd like to make the public nowcasting docs nicer looking and easier for new users to read. By moving from Swagger UI to ReDoc (also included by default in FastAPI) we get a nicer UI, with a sidebar nav which can have static pages in it in future. This PR is just focused on switching to ReDoc with a custom theme that aligns with our brand.

If we're happy with these new docs, we should change the route from /newdocs to /docs.

Note: I had to made some docker changes to run this repo locally. If docker is used in prod, you may want to throw the commit with those changes away or move them to renamed files for localdev.

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -132,3 +122,143 @@ async def get_api_information():
...
+                }
+            }
+        }""" + f""", document.getElementById('redoc-container'))
+    </script>
+    </body>
+    </html>
+    """
+    return HTMLResponse(html)
+
+@app.get("/newdocs", include_in_schema=False)
```
Comment (peterdudfield) [inline src/main.py:230]: change to "/docs"

File: src/main.py [hunk 2]
```diff
@@ -132,3 +122,143 @@ async def get_api_information():
 async def get_favicon() -> FileResponse:
     """Get favicon"""
     return FileResponse("src/favicon.ico")
+
+def get_redoc_html_with_theme(
```
Comment (peterdudfield) [inline src/main.py:126]: could move to a seperate `redoc_theme.py` file

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,15 @@
-FROM python:3.10-slim-buster
+FROM python:3.10-slim-bullseye
 
 # copy files over
-COPY ./src /app/src
+# COPY ./src /app/src
```
Comment (peterdudfield) [inline Dockerfile:4]: I think we need this back in

**Review by peterdudfield — CHANGES_REQUESTED** [review]
Looks really good @dctanner - thanks for this. 

Ive left a few comments which are minor. 

Would you also be able to add a screenshot to the PR? So we can easily see the look of the new docs?

I'm happy to solve conflicting files and help make CI work

File: Dockerfile (hunk 3 above)
Comment (dctanner) [reply Dockerfile:4]: volumes section in docker-compose.yml mounts the source now.

Comment (peterdudfield) [reply Dockerfile:4]: yea but not in production, perhaps I can put the Testing flag in there

**Comment by peterdudfield** [comment]
Thanks @dctanner I'll take it from here

**Comment by peterdudfield** [comment]
Actually @dctanner could you have another look at this? 
With the old docs I could try the route out, which I thought was a really useful function. Could you see if this can be included here?

**Comment by peterdudfield** [comment]
This blog seems to imply that Swagger is the better thing to use - https://www.wappalyzer.com/compare/redoc-vs-swagger-ui

File: src/main.py [hunk 4]
```diff
@@ -60,20 +64,7 @@
...
-        "name": "Open Climate Fix",
-        "url": "https://openclimatefix.org",
-        "email": "info@openclimatefix.org",
-    },
-    license_info={
-        "name": "MIT License",
-        "url": "https://github.com/openclimatefix/nowcasting_api/blob/main/LICENSE",
-    },
-)
+app = FastAPI(docs_url=None, redoc_url=None)
```
Comment (peterdudfield) [inline src/main.py:67]: docs_url could be "/swagger" then I (and maybe ohter people) can still access swagger docs if they want

**Comment by dctanner** [comment]
> This blog seems to imply that Swagger is the better thing to use - https://www.wappalyzer.com/compare/redoc-vs-swagger-ui

Given Swagger UI was made by the creators of OpenAPI, it's the default option for most OpenAPI things, and definitely the most used. It does feel like ReDoc has a more beginner friendly UI, and it aligns more with some of the best API docs like stripe etc.
Let's include /swagger as then there's the option at least.

**Pull request merged**